src/util/path/path.ts

```diff
@@ -697,48 +805,67 @@ export const getPathSegmentsInfo = (path: PathData): TPathSegmentsInfo[] => {
   return info;
 };
 
+/**
+ * Get the point on the path that is distance along the path
+ * @param path
+ * @param distance
+ * @param infos
+ */
 export const getPointOnPath = (
-  path: PathData,
+  path: TSimplePathData,
   distance: number,
-  infos: ReturnType<typeof getPathSegmentsInfo> = getPathSegmentsInfo(path)
-) => {
+  infos: TPathSegmentsInfo[] = getPathSegmentsInfo(path)
+): TPointAngle | undefined => {
   let i = 0;
   while (distance - infos[i].length > 0 && i < infos.length - 2) {
     distance -= infos[i].length;
     i++;
   }
-  // var distance = infos[infos.length - 1] * perc;
   const segInfo = infos[i],
     segPercent = distance / segInfo.length,
     command = segInfo.command,
     segment = path[i];
-  let info;
+  let info: TPointAngle;
 
-  switch (command) {
+  switch (segInfo.command) {
     case 'M':
       return { x: segInfo.x, y: segInfo.y, angle: 0 };
     case 'Z':
-    case 'z':
-      info = new Point(segInfo.x, segInfo.y).lerp(
-        new Point(segInfo.destX, segInfo.destY),
-        segPercent
-      );
+      info = {
+        ...new Point(segInfo.x, segInfo.y).lerp(
+          new Point(segInfo.destX, segInfo.destY),
+          segPercent
+        ),
+        angle: 0,
+      };
       info.angle = Math.atan2(
         segInfo.destY - segInfo.y,
         segInfo.destX - segInfo.x
       );
       return info;
     case 'L':
-      info = new Point(segInfo.x, segInfo.y).lerp(
-        new Point(segment[1], segment[2]),
-        segPercent
-      );
-      info.angle = Math.atan2(segment[2] - segInfo.y, segment[1] - segInfo.x);
+      if (isAbsLineCmd(segment)) {
+        info = {
+          ...new Point(segInfo.x, segInfo.y).lerp(
+            new Point(segment[1], segment[2]),
+            segPercent
+          ),
+          angle: 0,
+        };
+        info.angle = Math.atan2(segment[2] - segInfo.y, segment[1] - segInfo.x);
+      } else {
+        // TODO: provide a proper error
+        throw Error(
+          `Segment/path info mismatch, expected L, got ${segment[0]}`
+        );
+      }
```

Comment:
why this change is necessary?

Reply:
Cursed typing stuff again. If you don't verify it's a line, TS thinks it could be a z command which makes segment[1] be undefined. It SHOULD never happen, so the error is probably superfluous, but again I couldn't figure out how to make TS understand that in an easier way.